### PR TITLE
Improve error message when running on unsupported Java version

### PR DESCRIPTION
**Pull request by basil**
The error messages printed when running on an unsupported version of Java had a very poor user experience (UX): multiple message being printed for the same error, unnecessary mentioning of class versions, and unnecessary stack traces. This PR simplifies the error messages to the minimum needed to convey the necessary information to the user.

#### Java 8

```
$ java -jar war/target/jenkins.war
Running with Java 8 from /usr/lib/jvm/java-8-openjdk-amd64/jre, which is older than the minimum required version (Java 11).
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.
$ echo $?
1
```

#### Java 11 and 17

Started without logging anything special, as expected.

#### Java 18

```
$ java -jar war/target/jenkins.war                     
Running with Java 18 from /usr/lib/jvm/java-18-openjdk-amd64, which is not yet fully supported.
Run the command again with the --enable-future-java flag to enable preview support for future Java versions.
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.
$ echo $?
1
$ java -jar war/target/jenkins.war --enable-future-java
Sep 28, 2022 2:35:23 PM executable.Main verifyJavaVersion
WARNING: Running with Java 18 from /usr/lib/jvm/java-18-openjdk-amd64, which is not yet fully supported. Continuing because --enable-future-java is set. Supported Java versions are: [11, 17]. See https://jenkins.io/redirect/java-support/ for more information.
Running from: /home/basil/src/jenkinsci/jenkins/war/target/jenkins.war
webroot: $user.home/.jenkins
```

### Proposed changelog entries

Improve the error message when running the controller on unsupported Java version.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7185"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Comment by qwe1290670214** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: war/src/main/java/executable/Main.java
Comment (daniel-beck) [inline war/src/main/java/executable/Main.java:120]: This message might look a bit weird with Java 12…16 (phrased as if it's too new). If that's not a case we care about, that's fine.

**Review by daniel-beck — DISMISSED** [review]

File: war/src/main/java/executable/Main.java
Comment (basil) [reply war/src/main/java/executable/Main.java:118]: commit f296111fa7

```
$ java -jar war/target/jenkins.war
Running with Java 13 from /home/basil/Downloads/Direct/jdks/jdk-13.0.2+8, which is not fully supported.
Run the command again with the --enable-future-java flag to bypass this error.
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.

$  java -jar war/target/jenkins.war
Running with Java 18 from /home/basil/Downloads/Direct/jdks/jdk-18.0.2+9, which is not yet fully supported.
Run the command again with the --enable-future-java flag to enable preview support for future Java versions.
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.
```

**Pull request merged**
